File: sdks/csharp/src/NovaBilling/Payments/Requests/RefundPaymentDto.cs
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record RefundPaymentDto
{
    /// <summary>
    /// Payment ID
    /// </summary>
    [JsonIgnore]
    public required string Id { get; set; }

    /// <summary>
    /// Amount to refund (full refund if omitted)
    /// </summary>
    [JsonPropertyName("amount")]
    public double? Amount { get; set; }

    /// <summary>
    /// Reason for refund
    /// </summary>
    [JsonPropertyName("reason")]
    public string? Reason { get; set; }

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}

File: sdks/php/src/Payments/Requests/RefundPaymentDto.php
<?php

namespace NovaBilling\Payments\Requests;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class RefundPaymentDto extends JsonSerializableType
{
    /**
     * @var ?float $amount Amount to refund (full refund if omitted)
     */
    #[JsonProperty('amount')]
    public ?float $amount;

    /**
     * @var ?string $reason Reason for refund
     */
    #[JsonProperty('reason')]
    public ?string $reason;

    /**
     * @param array{
     *   amount?: ?float,
     *   reason?: ?string,
     * } $values
     */
    public function __construct(
        array $values = [],
    ) {
        $this->amount = $values['amount'] ?? null;
        $this->reason = $values['reason'] ?? null;
    }
}
